runtime/service/src/main/java/org/apache/polaris/service/task/TaskExecutorImpl.java

```diff
@@ -36,13 +44,15 @@
 import org.apache.polaris.service.events.AfterTaskAttemptedEvent;
 import org.apache.polaris.service.events.BeforeTaskAttemptedEvent;
 import org.apache.polaris.service.events.PolarisEventListener;
+import org.apache.polaris.service.tracing.TracingFilter;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Given a list of registered {@link TaskHandler}s, execute tasks asynchronously with the provided
  * {@link CallContext}.
  */
+@ApplicationScoped
 public class TaskExecutorImpl implements TaskExecutor {
```

Comment:
```suggestion
class TaskExecutorImpl implements TaskExecutor {
```

Probably similar for other implementations.

If it works, the constructors can be package-private as well.